src/datachain/lib/dc.py

```diff
@@ -831,8 +831,27 @@ def mutate(self, **kwargs) -> "Self":
         )
         ```
         """
-        chain = super().mutate(**kwargs)
-        chain.signals_schema = self.signals_schema.mutate(kwargs)
+        mutated = {}
+        schema = self.signals_schema
```

Comment:
It's not clear why the logic is separated between SignalSchema.mutete() and this code.
Ideally, we do all the logic in SignalSchema and DC just triggers this.
WDYT?

Reply:
That was my initial idea but the thing is that those 2 methods are doing different things:
1. `SignalsSchema.mutate(...)` only needs to update it's `values` object which is pretty straightforward. 
2. `DataChain.mutate(...)` must prepare fields that need change for underlying `DatasetQuery.mutate(...)` which doesn't have knowledge of pydantic objects...so if pydantic object is renamed it needs to gather all of it's sub-fields and send to underlying mutate. 

I think now that I simplified `DataChain.mutate()` it's decent but maybe this can be refactored even more, idk.